packages/api/src/controllers/asset.ts

```diff
@@ -527,23 +536,89 @@ app.post(
       asset
     );
 
-    res.json({ url, asset, task });
+    res.json({ url, tusEndpoint, asset, task });
   }
 );
 
-app.put("/upload/:url", async (req, res) => {
-  const {
-    params: { url },
-    config: { jwtSecret, jwtAudience },
-  } = req;
-  const { uploadUrl, playbackId } = parseUploadUrl(url, jwtSecret, jwtAudience);
+let tusServer: tus.Server;
 
-  const assets = await db.asset.find({ playbackId }, { useReplica: false });
-  if (!assets?.length || !assets[0]?.length) {
-    throw new NotFoundError(`asset not found`);
+export const setupTus = async (objectStoreId: string): Promise<void> => {
+  tusServer = await createTusServer(objectStoreId);
+};
+
+async function createTusServer(objectStoreId: string) {
+  const os = await db.objectStore.get(objectStoreId);
+
+  const url = new URL(os.url);
+  const [_, vodRegion, vodBucket] = url.pathname.split("/");
+  let protocol = url.protocol;
+  if (protocol.includes("+")) {
+    protocol = protocol.split("+")[1];
+  }
+
+  const opts: tus.S3StoreOptions | S3ClientConfig = {
+    path: "/upload/tus",
+    bucket: vodBucket,
+    accessKeyId: url.username,
+    secretAccessKey: url.password,
+    region: vodRegion,
+    partSize: 8 * 1024 * 1024,
+    tmpDirPrefix: "tus-tmp-files",
+    endpoint: "https://storage.googleapis.com",
+    namingFunction,
+  };
+  const tusServer = new tus.Server();
+  tusServer.datastore = new tus.S3Store(opts as tus.S3StoreOptions);
+  tusServer.on(tus.EVENTS.EVENT_UPLOAD_COMPLETE, onTusUploadComplete);
+  return tusServer;
+}
+
+export const setupTestTus = async (): Promise<void> => {
+  tusServer = await createTestTusServer();
+};
+
+async function createTestTusServer() {
+  const tusTestServer = new tus.Server();
+  tusTestServer.datastore = new tus.FileStore({
+    path: "/tmp",
```

Comment:
You're probably missing a `namingFunction` here as well. Keep in mind that you can't use the same one as below, since that one creates a file name with a directory, and the `FileStore` seemed to not like that when I tested it. You can just inline some function here like
```suggestion
    path: "/tmp",
    namingFunction: (req) => req.res.getHeader("livepeer-playback-id").toString(),
```

To keep the file name just equal to the playback ID. Then it's also easy on your tests to check if the file was created and is equal to what you sent. WDYT?